spec/models/manageiq/providers/redhat/infra_manager/provision/configuration_spec.rb

```diff
@@ -13,13 +13,43 @@
   before { allow_any_instance_of(ManageIQ::Providers::Redhat::InfraManager::Provision).to receive(:with_provider_destination).and_yield(provider_object) }
   before(:each) { allow_any_instance_of(ManageIQ::Providers::Redhat::InfraManager).to receive(:supported_api_versions).and_return([3]) }
   context "#attach_floppy_payload" do
-    it "should attach floppy if customization template provided" do
-      task.options[:customization_template_id] = cust_template.id
+    context "with ovirt gem" do
+      it "should attach floppy if customization template provided" do
+        task.options[:customization_template_id] = cust_template.id
 
-      expect(task).to     receive(:prepare_customization_template_substitution_options).and_return('key' => 'value')
-      expect(rhevm_vm).to receive(:attach_floppy).with(cust_template.default_filename => '#some_script')
+        expect(task).to     receive(:prepare_customization_template_substitution_options).and_return('key' => 'value')
+        expect(rhevm_vm).to receive(:attach_floppy).with(cust_template.default_filename => '#some_script')
+
+        task.attach_floppy_payload
+      end
+    end
+
+    context "with ovirtsdk4" do
+      let(:vm_proxy)        { OvirtSDK4::Vm.new }
+      let(:vm_service)      { double("OvirtSDK4::VmService", :get => vm_proxy) }
+      let(:connection)      { double("OvirtSDK4::Connection") }
+      let(:ovirt_services)  { double("ManageIQ::Providers::Redhat::InfraManager::OvirtServices::Strategies::V4") }
+      let(:provider_object) { ManageIQ::Providers::Redhat::InfraManager::OvirtServices::Strategies::V4::VmProxyDecorator.new(vm_service, connection, ovirt_services) }
+
+      it "updates the vm with the right payload" do
+        task.options[:customization_template_id] = cust_template.id
+        #allow(vm_service).to receive(:get).and_return()
```

Comment:
Why do we need this commented line?

Reply:
what commented line ,which I totally did not forget to remove, do you mean? ;)

Reply:
Line 36 `#allow(vm_service).to receive(:get).and_return()`

Reply:
@pkliczewski done :)